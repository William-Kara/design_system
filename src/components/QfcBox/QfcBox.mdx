import { ArgTypes, Meta } from '@storybook/blocks';
import { Story } from '@storybook/addon-docs';
import QfcBox from '@/components/QfcBox/QfcBox.vue';

<Meta title="Elements/Box" />

# Definition

## Props

The following props are available for the QfcBox component:

<ArgTypes of={QfcBox} />

# Usage

To use QfcBox, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/bulma/elements/box.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcBox.scss';
```

```html
<template>
  <QfcBox size="medium">
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit.</p>
  </QfcBox>
</template>
```

# Declinaison

## FullWidth

<Story name="Primary" id="elements-box--primary" />

## Medium

<Story name="Medium" id="elements-box--medium" />
