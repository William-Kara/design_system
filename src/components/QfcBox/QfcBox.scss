.box {
  overflow: hidden;
  box-sizing: border-box;
  margin-bottom: 0 !important;

  &.--small {
  }
  &.--medium {
    width: 27.4rem;
    height: 32rem;
  }
  &.--large {
  }
}
.--padding-small {
    padding: $q-generic-space-4 !important;
}
