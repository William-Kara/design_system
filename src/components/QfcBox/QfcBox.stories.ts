import Box from './QfcBox.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Elements/Box',
  argTypes: {
    size: {
      type: { name: 'string', required: false },
      defaultValue: 'fullwidth',
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    size: 'fullwidth',
  },
};

export const Medium: Story = {
  args: {
    size: 'medium',
  },
};
