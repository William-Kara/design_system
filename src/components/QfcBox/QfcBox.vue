<template>
  <div
    :class="{
      box: true,
      '--small': size === 'large',
      '--medium': size === 'medium',
      '--large': size === 'small',
      '--padding-small': padding === 'small',
    }"
  >
    <slot />
  </div>
</template>

<script lang="ts" setup>
const sizeType = ['small', 'medium', 'large', 'fullwidth'] as const;
type SizeType = (typeof sizeType)[number];

interface Props {
  size: SizeType;
  padding: SizeType;
}
withDefaults(defineProps<Props>(), {
  size: 'fullwidth',
  padding: 'medium',
});
</script>

<style lang="scss" scoped src="./QfcBox.scss" />
