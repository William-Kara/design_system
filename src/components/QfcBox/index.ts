import QfcBox from './QfcBox.vue';

export { QfcBox };
export default QfcBox;
