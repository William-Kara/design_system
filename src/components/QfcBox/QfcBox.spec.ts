import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcBox from './QfcBox.vue';

let wrapper: any = null;

const propsValues = {
  size: 'fullwidth',
  padding: 'medium',
};

beforeEach(() => {
  wrapper = shallowMount(QfcBox, {
    propsData: propsValues,
  });
});

describe('QfcBox Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
