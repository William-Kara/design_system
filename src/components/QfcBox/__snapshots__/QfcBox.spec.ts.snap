// Vitest Snapshot v1

exports[`QfcBox Component > should render correctly 1`] = `"<div data-v-e72cc035=\\"\\" class=\\"box\\"></div>"`;
